src/hera/workflows/steps.py

```diff
@@ -62,6 +65,52 @@ def finished_at(self) -> str:
     def result(self) -> str:
         return f"{{{{steps.{self.name}.outputs.result}}}}"
 
+    def get_parameters_as(self, name):
+        """Gets all the output parameters from this task"""
+        return Parameter(name=name, value_from=ValueFrom(expression=f"{{{{steps.{self.name}.outputs.parameters}}}}"))
+
+    def get_parameter(self, name: str) -> Parameter:
+        """Returns a Parameter from the task's outputs based on the name.
+
+        Parameters
+        ----------
+        name: str
+            The name of the parameter to extract as an output.
+
+        Returns
+        -------
+        Parameter
+            Parameter with the same name
+        """
+        if isinstance(self.template, str):
+            raise ValueError(f"Cannot get output parameters when the template was set via a name: {self.template}")
+
+        # here, we build the template early to verify that we can get the outputs
+        if isinstance(self.template, Templatable):
+            template = self.template._build_template()
+        else:
+            template = self.template
+
+        # at this point, we know that the template is a `Template` object
+        if template.outputs is None:  # type: ignore
+            raise ValueError(f"Cannot get output parameters when the template has no outputs: {template}")
+        if template.outputs.parameters is None:  # type: ignore
+            raise ValueError(f"Cannot get output parameters when the template has no output parameters: {template}")
+        parameters = template.outputs.parameters  # type: ignore
+
+        obj = next((output for output in parameters if output.name == name), None)
+        if obj is not None:
+            if isinstance(obj, _ModelParameter):
+                obj.value = f"{{{{steps.{self.name}.outputs.parameters.{name}}}}}"
+                return Parameter(
+                    name=obj.name,
+                    value=obj.value,
+                    value_from=obj.value_from,
+                    global_name=obj.global_name,
+                    description=obj.description,
+                )
```

Comment:
@flaviuvadan I am not sure what this code is doing. Can you explain a bit on what is happening here?

Reply:
I removed the check for `_ModelParameter` because it's not useful - all the objects in parameters are of type `_ModelParameter` because the code invokes `build_template` just above. Here we assemble the identified output parameter and return a Hera `Parameter` type.